### DISCUSSION TITLE
Install in windows instructions and bonus tips

### DISCUSSION THREAD

**Discussion by foxbg** [danielmiessler/Fabric#263] [General]
Here are the steps I've used to install in windows. No WSL required.

Source: https://labs.hbcom.info/posts/fabric/

Compressed to commands only for easier view...
Open PowerShell and type
```
cd /path/to/your/code
git clone https://github.com/danielmiessler/fabric.git
py.exe -m venv fabric
cd fabric
.\Scripts\activate
pip install .
fabric --setup
```
Restart Your Shell
`fabric --help`

Enjoy :)

pbpaste is Get-Clipboard in powershell.

**Comment by eerondamoon** [comment]


**Reply by foxbg** [reply]
Are you using PowerShell or something else? From the error it looks like you might be using WSL or Linux.

**Reply by eerondamoon** [reply]
Yes i'm using wsl2. Your instruction work for PowerShell, I just mix between your instruction and the official one. Sry

**Reply by foxbg** [reply]
For WSL follow the official instructions. I've updated the initial post to clearly indicate it.

**Comment by Sluengo** [comment]
You saved me a bunch of time man. I was trying to follow the "official" instructions using wsl. I kept running into issues with my python version. I suppose the dependencies that the project uses requires version 10 of python or greater. I was going through hopes on windows trying to get an upgraded version. These instructions you provided just work on Windows.

I'm going to ask them to update their readme to include a note to make sure you have at least python 3.10 installed.

**Comment by renmiller** [comment]
I'm so sorry to bother, but could you point me in the direction of where I should start if I'm basically computer illiterate. Are you aware of any step-by-step guide written for smoothbrains such as myself?

**Comment by pbaruns** [comment]
@foxbg this is great! thanks for giving Windows users a right direction. Could you please kindly let me know if there is any free API that allows to be used with Fabric? From what I understand, OpenAI no longer provides free API keys. Which API key service do you suggest that comes free just to experience Fabric?

**Reply by foxbg** [reply]
I can suggest you to use locally hosted model. You are in control of the data and the cost.

**Reply by pbaruns** [reply]
Thanks so much for the quick response! 
I don't know how to go about doing that (hosting local models), but you got me in the right direction, again! thanks friend!

**Comment by anthonyonazure** [comment]
I am getting an error when trying to install with pip, any ideas? Thanks

\Source\fabric\fabric> pip install .
ERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.

**Reply by anthonyonazure** [reply]
I just created a setup.py file and ran the pip install . and this was the output
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: fabric
  Building wheel for fabric (pyproject.toml) ... done
  Created wheel for fabric: filename=fabric-0.1.0-py3-none-any.whl size=922 sha256=f46e8048c570778630888f06e14cee1c7d5e1cf25153dc0212ebc08d0a6b66c0
  Stored in directory: C:\Users\antho\AppData\Local\Temp\pip-ephem-wheel-cache-ncqeq21e\wheels\6f\03\1c\40320afec03f8db90ba8f42ce329babf8842af17c68e7a716b
Successfully built fabric
Installing collected packages: fabric
Successfully installed fabric-0.1.0

**Comment by foxbg** [comment]
Have in mind that fabric is now go package. As noted in the documentation (https://github.com/danielmiessler/fabric?tab=readme-ov-file#Installation) all previous python installation instructions are not valid for newer releases.
